File: MyBudget-back/src/main/java/com/example/mybudget/models/dto/Account.java
package com.example.mybudget.models.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Account implements Serializable {
    private Integer id;
    private String name;
    private String currency;
    private BigDecimal balance;
    @JsonIgnore
    private List<Transaction> transactions;
}

File: MyBudget-front/src/app/models/account.ts
import { Transaction } from './transaction';

export class Account {
  id: number;
  name: string;
  currency: string;
  balance: number;
  transactions: Transaction[];

  constructor(
    id: number,
    name: string,
    currency: string,
    balance: number,
    transactions: Transaction[]
  ) {
    this.id = id;
    this.name = name;
    this.currency = currency;
    this.balance = balance;
    this.transactions = transactions;
  }
}
